concepts/operator-reconciliation/README.md

```diff
@@ -0,0 +1,150 @@
+2. Decide a common approach to make sure that we are able to abstract the Operator communication not only for interaction with a local cluster (the cluster where the operator is running on) but also a remote cluster
+3. Make sure that we can safely watch all necessary Kyma Resources without much CPU or Memory Overhead for a Reconciler
+4. Make sure that our Developer Experience does not suffer from the new approach. Ideally, we want to be able to take over Component Reconciler Logic and keep the interfaces as stable as possible
+
+A derived action plan should tackle the following topics (from first to last):
+
+1. Create a decision document on which Operator Framework to use / not to use (e.g. kubebuilder, operator-sdk, native) and why (can be running document and updated regularly on changes)
+2. Spin-Up a Bootstrap Operator Repository for a Component Reconciler in which we can work further, communicate existance of POC goals in Kyma teams to raise awareness and prompt for collaborators
+3. Create an interface similar to our current K8s client in the reconciler that can deal with in-cluster and remote cluster API-Server Access
+4. Create an implementation that is able to deal with deploying and updating Helm Chart data (templated charts as native resources)
+5. Create a suggestion for a separate Provisioning Operator that can deploy CRDs and the Component Operator
+6. Derive a final Low-Level Architecture that suggests a scaled implementation while making sure to keep team responsibility separate
+
+## High level architecture
+
+For Operator-based reconciliations, we want to enable maximum flexibility of the architecture depending on the needs of the Kyma user. This means that we want to allow
+
+1. Fully Independent Small-Scale Clusters able to reconcile themselves, relying on external node setup (e.g. running for a local evaluation in k3d or for smaller scale deployments)
+2. Fully centralized Control Plane Configurations, in which the Control Plane is not only provisioning the cluster, but also hosting the reconciliation, ensuring minimal overhead in the workload cluster
+3. Lightweight Control Plane Configurations (Hybrid Mode), in which the Control Plane is able to provision the cluster and create/update an inventory of all managed cluster, but not taking care of reconciliation
+
+The main focus of the POC will be to prove the functionality of the Reconciliation Operator to work with all 3 modes independant of the final implementation. The POC needs to prove that we are able to reconcile a cluster remotely or in-cluster without a significant difference in reconciliation logic.
+
+![high_level_view_operator-reconciliation](assets/high_level_view_operator-reconciliation.png)
+
+### Single-Cluster (Small-Scale)
+
+![single_cluster_configuration](assets/single_cluster_configuration.png)
```

Comment:
Good to point out these two different modes: small-scale and centralised. But I could imaging that we should put both in one diagram to make also obvious, that the differences are not that huge:

* The main difference is, that in the centalised deployment will be many Kyma-CRs, Cluster-CRs etc. But in the small-scale mode will exist only one of them pointing to the local cluster.

Reply:
Big bad diagram will contain only one visualization with a disclaimer for small-scale